### PR TITLE
Add EIP-5267: Retrieval of EIP-712 domain

### PR DESCRIPTION
**Pull request by frangio**

### CODE REVIEW

File: EIPS/eip-draft_retrieval_domain.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
+---
+eip: <to be assigned>
```
Comment (MicahZoltu) [inline EIPS/eip-draft_retrieval_domain.md:2]:
```suggestion
eip: 5267
```

Comment (Pandapip1) [reply EIPS/eip-draft_retrieval_domain.md:2]: image
@SamWilsn how about having a custom error specifically when the EIP "number" is set to `<to be assigned>`?

File: EIPS/eip-draft_retrieval_domain.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+title: Retrieval of EIP-712 domain
+description: A way to describe and retrieve an EIP-712 domain to securely integrate EIP-712 signatures.
+author: Francisco Giordano (@frangio)
+discussions-to: <URL>
```
Comment (Pandapip1) [inline EIPS/eip-draft_retrieval_domain.md:6]: Should be a thread on https://ethereum-magicians.org/

**Comment by xinbenlv** [comment]
I feel this is going to be extremely useful. @frangio thank you for proposing. Left some questions about the design decisions made/implied in the proposal.

File: EIPS/eip-draft_retrieval_domain.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+Compliant contracts MUST define `eip712Domain` exactly as declared below. All specified values MUST be returned even if they are not used, to ensure proper decoding on the client side.
+
+```solidity
+function eip712Domain() external view returns (
+    bytes1 fields,
+    string name,
+    string version,
+    uint256 chainId,
+    address verifyingContract,
+    bytes32 salt,
```
[the remark this answers is not in the document]
Comment (frangio) [reply EIPS/eip-draft_retrieval_domain.md]: I added a note in https://github.com/ethereum/EIPs/pull/5267/commits/664ccf69148c8fae6a6e874f569c5fd0eefc8587

File: EIPS/eip-5267.md [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+eip: 5267
+title: Retrieval of EIP-712 domain
+description: A way to describe and retrieve an EIP-712 domain to securely integrate EIP-712 signatures.
+author: Francisco Giordano (@frangio)
+discussions-to: https://ethereum-magicians.org/t/eip-5267-retrieval-of-eip-712-domain/9951
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-07-14
+requires: 712
```
Comment (Pandapip1) [inline EIPS/eip-5267.md:11]:
```suggestion
requires: 155, 712, 2612
```
```suggestion
requires: 712
```

File: EIPS/eip-5267.md [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+User-agents or applications that implement this EIP SHOULD additionally support those contracts that due to their immutability cannot be upgraded to implement this EIP, by hardcoding their domain based on contract address and chain id.
+
+## Reference Implementation
+
+### Solidity Example
+
+```solidity
+pragma solidity 0.8.0;
+
+contract Example {
```
Comment (Pandapip1) [inline EIPS/eip-5267.md:69]: Please use a more descriptive name.

**Review by Pandapip1 — CHANGES_REQUESTED** [review]

File: EIPS/eip-5267.md (hunk 4 above)
Comment (frangio) [reply EIPS/eip-5267.md:11]: What is the semantics of the requires field? EIP-2612 is definitely not a requirement, it's just used as an example. EIP-155 is arguably not a requirement either as the chainId doesn't need to be in use.

**Review by Pandapip1 — APPROVED** [review]

**Pull request merged**